File: PHP/PHP Basics/16.Encapsulation and Inheritance - Exercises/Part I - Encapsulation/01.Box/Box.php
<?php

class Box
{
    private const MESSAGE_ERROR_LENGTH = 'Length cannot be zero or negative.';
    private const MESSAGE_ERROR_WIDTH = 'Width cannot be zero or negative.';
    private const MESSAGE_ERROR_Height = 'Height cannot be zero or negative.';

    private $length;
    private $width;
    private $height;

    public function __construct($length, $width, $height)
    {
        $this->setLength($length);
        $this->setWidth($width);
        $this->setHeight($height);
    }

    public function getLength()
    {
        return $this->length;
    }

    public function getWidth()
    {
        return $this->width;
    }

    public function getHeight()
    {
        return $this->height;
    }

    public function surfaceArea() {
        //formula: 2lw + 2lh + 2wh;
        return round(2 * $this->getLength() * $this->getWidth() + 2
            * $this->getLength() * $this->getHeight() + 2 * $this->getWidth()
            * $this->getHeight(), 2);
    }

    public function lateralSurfaceArea() {
        //formula: 2lh + 2wh
        return round(2 * $this->getLength() * $this->getHeight() + 2
            * $this->getWidth() * $this->getHeight() , 2);
    }

    public function volume() {
        //formula: lwh
        return round($this->getLength() * $this->getWidth() * $this->getHeight() , 2);
    }

    private function setLength($length)
    {
        if ($length <= 0) {
            throw new Exception(self::MESSAGE_ERROR_LENGTH);
        }

        $this->length = $length;
    }

    private function setWidth($width)
    {
        if ($width <= 0) {
            throw new Exception(self::MESSAGE_ERROR_WIDTH);
        }

        $this->width = $width;
    }

    private function setHeight($height)
    {
        if ($height <= 0) {
            throw new Exception(self::MESSAGE_ERROR_Height);
        }

        $this->height = $height;
    }

    public function __toString()
    {
       $builder = sprintf("Surface Area - %.2f\n", $this->surfaceArea());
       $builder .= sprintf("Lateral Surface Area - %.2f\n", $this->lateralSurfaceArea());
       $builder .= sprintf("Volume - %.2f\n" , $this->volume());

       return $builder;
    }
}

$length = readline();
$width = readline();
$height = readline();

try {
    $box = new Box($length, $width, $height);
    echo $box;
} catch (Exception $e) {
    echo $e->getMessage() . PHP_EOL;
}

File: C# Advanced/C# OOP/06.Encapsulation - Exercise/03. CSharp-OOP-Encapsulation-Exercise-AnimalFarm/AnimalFarm/01.ClassBox/Box.cs
using System;

namespace ClassBox
{
    public class Box
    {
        private double length;
        private double width;
        private double height;

        public Box(double length, double width, double height)
        {
            this.Length = length;
            this.Width = width;
            this.Height = height;
        }

        public double Length
        {
            get
            {
                return this.length;
            }
            private set
            {
                if (InvalidInputValue(value))
                {
                    throw new ArgumentException("Length cannot be zero or negative.");
                }

                this.length = value;
            }
        }

        public double Width
        {
            get
            {
                return this.width;
            }
            private set
            {
                if (InvalidInputValue(value))
                {
                    throw new ArgumentException("Width cannot be zero or negative.");
                }
                this.width = value;
            }
        }

        public double Height
        {
            get
            {
                return this.height;
            }
            private set
            {
                if (InvalidInputValue(value))
                {
                    throw new ArgumentException("Height cannot be zero or negative.");
                }
                this.height = value;
            }
        }

        public double CalculateVolume()
        {
            var volume = this.length * this.width * this.height;
            return volume;
        }

        public double CalculateSurfaceArea()
        {
            var area = 2 * length * width + 2 * length * height + 2 * width * height;
            return area;
        }

        public double CalculateLateralSurfaceArea()
        {
            var area = 2 * length * height + 2 * width * height;
            return area;
        }

        private static bool InvalidInputValue(double value)
        {
            return value <= 0;
        }
    }
}
